src/index.ts

```diff
@@ -1,4 +1,4 @@
-import { proxy, subscribe, getVersion } from 'valtio/vanilla';
```

Comment:
Its really interesting that you removed the proxy completely. Do you know why it was unnecessary?

Reply:
It was used to recursively bind and subscribe to child objects and arrays. Now that we're only subscribing to the top level proxy, we shouldn't need to create new child proxies.

Reply:
Oh, so that means if you detach a child proxy, it will be unbound from yjs. Hm, I think its reasonable, different from the original design though. That would simplify things.

Reply:
Ah, hm, I'm not familiar with child proxies in valtio or what the desired behavior would be then. I'll have to check them out.

Reply:
That would mean the stuff mentioned here: https://github.com/pmndrs/valtio/blob/main/docs/how-tos/how-to-split-and-compose-states.mdx wouldn't be subscribed, because the only listener is on the original proxy, correct?

It feels somewhat wrong to me to diverge from valtios feature set, because it makes understanding the library more difficult. In my opinion all valtio features should just work.

In my refactor I found that it would be helpful to have more generic methods around working with yjs. One such functionality is to get/set something in the yDoc by path. It would be possible to build something on top of this to bind a nested a yValue to a nested part of the proxy. Not as simple as just using the proxy, but possible.

Reply:
Ah, that is what's meant by "detach a child proxy"? I believe that should all work fine as I understand valtio's subscribe captures events in child proxies.

Not sure how circular states behave though.

Reply:
And the other way around? Would an update to a yjs value update the detached child proxy? I think that's where we would lose the connection.

Reply:
here's an example - https://codesandbox.io/s/react-forked-2998r7

Reply:
@stephenarra By detaching a child proxy, I mean:
```js
const p = proxy({ child: { count: 1 } })
const p2 = p.child
delete p.child
// p2 is still a valid valtio proxy
```
I don't think there's much use case with yjs. What do you think?

> Not sure how circular states behave though.

yjs doesn't support circular structure anyway.

Reply:
Thanks for the clarification.

I agree, I don't think there's much of a use case with a single yjs document. Given `p` was bound to a yjs document, I would expect `child` to be removed and for updates to p2 not to recreate the object on the document. You should be able to bind `p2` to a new yjs document though. I believe this is consistent with the previous behavior.